draft-ietf-quic-qlog-main-schema.md

```diff
@@ -592,68 +592,86 @@ JSON serialization:
 
-The "time" field indicates the timestamp at which the event occurred. Its value is
-typically the Unix timestamp since the 1970 epoch (number of milliseconds since
-midnight UTC, January 1, 1970, ignoring leap seconds). However, qlog supports two
-more succinct timestamps formats to allow reducing file size. The employed format
-is indicated in the "time_format" field, which allows one of three values:
-"absolute", "delta" or "relative".
+Each event MUST include a "time" field to indicate the timestamp that it
+occurred. The format of the value is governed by the TimeFormat, which MAY be
+included in the "time_format" field.
+
+There is no requirement that events in the same trace use the same time format.
+However, using a single time format for related events can make them easier to
+analyze.
 
 Definition:
 
 ~~~ cddl
-TimeFormat = "absolute" / "delta" / "relative"
+TimeFormat = "absolute" /
+             "delta" /
+             "relative" /
+             "monotonic" .default "absolute"
 ~~~
 {: #time-format-def title="TimeFormat definition"}
 
-* Absolute: Include the full absolute timestamp with each event. This approach
-  uses the largest amount of characters. This is also the default value of the
-  "time_format" field.
-* Delta: Delta-encode each time value on the previously logged value. The first
-  event in a trace typically logs the full absolute timestamp. This approach uses
-  the least amount of characters.
-* Relative: Specify a full "reference_time" timestamp (typically this is done
-  up-front in "common_fields", see {{common-fields}}) and include only
-  relatively-encoded values based on this reference_time with each event. The
-  "reference_time" value is typically the first absolute timestamp. This approach
-  uses a medium amount of characters.
-
-The first option is good for stateless loggers, the second and third for stateful
-loggers. The third option is generally preferred, since it produces smaller files
-while being easier to reason about. An example for each option can be seen in
+absolute:
+: A full absolute timestamp. This approach uses the
+  largest amount of characters. This is the default value of the "time_format"
+  field.
+
+delta:
+: A delta-encoded value, based on the previously logged value. The first event
+  in a trace typically logs the full absolute timestamp. This approach uses the
+  least amount of characters.
+
+relative:
+: A relative timestamp based on some relative time base.  The log SHOULD specify
+  a "reference_time" timestamp, either by providing it explicitly in the
+  "reference_time" field in "common_fields" ({{common-fields}}) or logging an
+  absolute time in the first event. This approach uses a medium amount of
+  characters.
+
+monotonic:
+: A relative timestamp based on an opaque monotonically nondecreasing clock.
+  The "reference_time" field MUST NOT be applied to this format, since it based
+  on an opaque clock that cannot be interpreted beyond the process that
```

Comment:
This whole sentence is hard to understand (I'm not even sure I really grasp it), and the `it's` is ambiguous: is it "this format" or "the reference_time"?

Overall, imo this section does not really explain how monotonic is different from relative, nor why it's useful/needed.

Reply:
I tried to clarify the sentence to make it clear what it being measured.

Beyond that, I'm not sure how to more clearly describe that `relative` and `monotonic` are different. The monotonic description is succinct. So the relative format probably needs more wordsmithing for preciseness, rather than have people making assumptions.